### PR TITLE
OOP school library: refactor your code

### PR DESCRIPTION
**Pull request by DaveZag**
# OOP school library: refactor your code
### On this branch we:
- Divide all the methods of the app.rb file into smaller modules.
- Create modules to handle book, person, and rental operations.
- Include these modules in the app.rb file.
- Refactor some parts of the code for best practices.

### CODE REVIEW

File: modules/operations/books_ops.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+require_relative '../book'
+
+module BookOperations
+  def list_books
+    if @books.empty?
+      puts 'No book found. Please insert one.'
+    else
+      @books.each_with_index do |book, index|
+        puts "#{index}) Title: \"#{book.title}\", Author: #{book.author} "
```
Comment (Olamarx) [inline modules/operations/books_ops.rb:9]: - [x] This book displays some data that should not be printed to the user. This does not happen to the book alone, but to the rentals also. The person creation works fine. I would advice you to fix this bug for a better user experience.

image

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters passed ✔️
- All functionalities ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: modules/operations/books_ops.rb (hunk 1 above)
Comment (DaveZag) [reply modules/operations/books_ops.rb:9]: @Olamarx Thanks for the feedback.
The requested changes have been implemented. 🙃

Comment (mckent05) [reply modules/operations/books_ops.rb:9]: **OPTIONAL** - The issue raised by the previous reviewer still persists, I think you forgot to push the changes made. Please ensure you push the changes to fix this issue 🙏

**Review by mckent05 — APPROVED** [review]
Hi team, 

# STATUS: APPROVED ✔️ ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations and well done! 🎉 
approved


## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👏 👏 
- App works as it should 👍 👍 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by DaveZag** [comment]
@mckent05 Thanks for pointing that out. It seems like I had a network problem but now it's fine and I've pushed the changes.

**Pull request merged**